### ISSUE TITLE
Uyuni 2022.03 - Cannot Bootstrap Servers - SaltERROR com.suse.manager.webui.controllers.utils.AbstractMinionBootstrapper 

### ISSUE THREAD

**Issue by spartonicus** [uyuni-project/uyuni#5136]
# Problem description

Upgraded to 2022.03 from 2022.02 and unable to bootstrap any new server.   Existing servers are all manageable just fine.  Rolling back to 2022.02 the same servers bootstrap using the same server, ip and credentials.   Servers are unique in terms machine_id and SSH keys (aka it works on 2022.02.  

## Version of Uyuni Server and Proxy (if used)

Information for package Uyuni-Server-release:
---------------------------------------------
Repository     : uyuni-server-stable
Name           : Uyuni-Server-release
Version        : 2022.03-172.4.uyuni1
Arch           : x86_64
Vendor         : obs://build.opensuse.org/systemsmanagement:Uyuni
Support Level  : Level 3
Installed Size : 1.4 KiB
Installed      : Yes (automatically)
Status         : out-of-date (version 2022.02-171.1.uyuni1 installed)
Source package : Uyuni-Server-release-2022.03-172.4.uyuni1.src
Summary        : Uyuni Server
Description    :
    Uyuni lets you efficiently manage physical, virtual,
    and cloud-based Linux systems. It provides automated and cost-effective
    configuration and software management, asset management, and system
    provisioning.

## Details about the issue

Issue is repeatable trying to bootstrap anything post-upgrade.   Followed the release notes and completed a `mgr-create-bootstrap-repo` post-upgrade.  No change following the bootstrap process. 

Output of  /var/log/rhn/rhn_web_ui.log (during bootstrap): 

`
2022-04-01 15:43:15,668 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-7] ERROR com.suse.manager.webui.controllers.utils.AbstractMinionBootstrapper - Error during bootstrap: SaltSSHError(255, Authorized uses only. All activity may be monitored and reported.
)
2022-04-01 15:51:23,203 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-10] ERROR com.suse.manager.webui.controllers.utils.AbstractMinionBootstrapper - Error during bootstrap: SaltSSHError(255, Authorized uses only. All activity may be monitored and reported.
)
2022-04-01 15:56:29,251 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-4] ERROR com.suse.manager.webui.controllers.utils.AbstractMinionBootstrapper - Error during bootstrap: SaltSSHError(255, Authorized uses only. All activity may be monitored and reported.
)
`
Output of /var/log/salt/api (during bootstrap):

`
2022-04-01 15:28:46,445 [salt.client.ssh  :1084][ERROR   ][8098] Error running ssh_pre_flight script preflight.sh
2022-04-01 15:43:11,162 [salt.roster      :106 ][ERROR   ][2279] Can't access roster for backend flat: Roster file "susemanager-roster-6251445279328521778.tmp" not found
2022-04-01 15:43:15,620 [salt.client.ssh  :1084][ERROR   ][9357] Error running ssh_pre_flight script preflight.sh
2022-04-01 15:47:04,889 [salt.utils.parsers:1106][WARNING ][2177] SaltAPI received a SIGTERM. Exiting.
2022-04-01 15:51:23,155 [salt.client.ssh  :1084][ERROR   ][4726] Error running ssh_pre_flight script preflight.sh
2022-04-01 15:56:25,416 [salt.roster      :106 ][ERROR   ][2237] Can't access roster for backend flat: Roster file "susemanager-roster-13628352508781928198.tmp" not found
2022-04-01 15:56:27,833 [salt.client.ssh  :1084][ERROR   ][5786] Error running ssh_pre_flight script preflight.sh
`

**Comment by sleepygh0st** [comment]
I can confirm the same behavior in my environment after upgrading to 2022.03

**Comment by juliogonzalez** [comment]
@vzhestkov can you have a look at this on Monday?

I thought it was maybe related to https://github.com/uyuni-project/uyuni/pull/5128, but doesn't seem to be the case.

**Comment by vzhestkov** [comment]
@spartonicus please check the logs on the client.
The message is from the client: `Authorized uses only. All activity may be monitored and reported.`
Is it possible that copying files with SSH is blocked somehow?

**Comment by sboger** [comment]
Brand new install today on fully updated openSUSE Leap 15.3 and Uyuni release 2022.03. Same exact behavior attempting Bootstrap from webui with centos7 and almalinux8 test vms. 

`
2022-04-02 16:51:37,901 [ajp-nio-127.0.0.1-8009-exec-7] ERROR com.suse.manager.webui.controllers.utils.AbstractMinionBootstrapper - Error during bootstrap: SaltSSHError(255, )
2022-04-02 16:51:45,379 [ajp-nio-127.0.0.1-8009-exec-2] ERROR com.suse.manager.webui.controllers.utils.AbstractMinionBootstrapper - Error during bootstrap: SaltSSHError(255, )
`

**Comment by vzhestkov** [comment]
What is the version of `salt` packages installed on the server. Please show the latest changelog entries also.

**Comment by vzhestkov** [comment]
Could you please also try to repeat the test after adding
```
ssh_use_salt_thin = true
```
To `/etc/rhn/rhn.conf`
And restarting `salt-secrets-config` service.

**Comment by gabira2003** [comment]
I can confirm the same behavior in my environment after upgrading to 2022.03
@vzhestkov , no improvement seen as well when applied your suggested fix.  ( setting ssh_use_salt_thin = true and running restart salt-secrets-config)

**Comment by vzhestkov** [comment]
Sorry namespace was missing in the parameter, should be
```
web.ssh_use_salt_thin = true
```
And repeat restarting `salt-secrets-config` after modifying the config.

**Comment by gabira2003** [comment]
> Sorry namespace was missing in the parameter, should be
> 
> ```
> web.ssh_use_salt_thin = true
> ```
> 
> And repeat restarting `salt-secrets-config` after modifying the config.

No improvement , same result :(

**Comment by vzhestkov** [comment]
Try to reastart tomcat also.

**Comment by gabira2003** [comment]
> Try to reastart tomcat also.

I even have restarted whole server.No success.

**Comment by vzhestkov** [comment]
Could you please show the latest changelog entries for python3-salt package? And what are the messages in /var/log/salt/api on attempt to bootstrap new client?

**Comment by sboger** [comment]
> Could you please show the latest changelog entries for python3-salt package? And what are the messages in /var/log/salt/api on attempt to bootstrap new client?

[python3-salt changelog](https://gist.github.com/sboger/141f819c3149c24196cb81c39818d083)

**Comment by sboger** [comment]
zypper se -v python3-salt
Loading repository data...
Reading installed packages...

S | Name         | Type  | Version               | Arch   | Repository
--+--------------+-------+-----------------------+--------+-------------------------------------------------------------
v | python3-salt | pac-> | 3002.2-8.41.8.1       | x86_64 | Main Repository
    name: python3-salt
i | python3-salt | pac-> | 3002.2-150300.53.10.1 | x86_64 | Update repository with updates from SUSE Linux Enterprise 15
    name: python3-salt
v | python3-salt | pac-> | 3002.2-150300.53.7.2  | x86_64 | Update repository with updates from SUSE Linux Enterprise 15
    name: python3-salt
v | python3-salt | pac-> | 3002.2-50.1.12.1      | x86_64 | Update repository with updates from SUSE Linux Enterprise 15
    name: python3-salt
v | python3-salt | pac-> | 3002.2-50.1.15.1      | x86_64 | Update repository with updates from SUSE Linux Enterprise 15
    name: python3-salt
v | python3-salt | pac-> | 3002.2-50.1.9.1       | x86_64 | Update repository with updates from SUSE Linux Enterprise 15
    name: python3-salt
v | python3-salt | pac-> | 3002.2-53.4.1         | x86_64 | Update repository with updates from SUSE Linux Enterprise 15
    name: python3-salt
v | python3-salt | pac-> | 3002.2-8.41.8.1       | x86_64 | Update repository with updates from SUSE Linux Enterprise 15
    name: python3-salt

**Comment by vzhestkov** [comment]
Thanks @sboger, just checked on my test system also. It seems that the latest change we've promoted to `salt` was not published yet, but the new functionality of using the Salt Bundle with Salt SSH is missing this feature from `salt`.
Please try to set the followin to `/etc/rhn/rhn.conf`:
```
web.ssh_salt_pre_flight_script = /usr/share/susemanager/salt-ssh/non-existing.sh
```
The value should be set to not-existing file path.

The following setting is not working as it relies on the feature mentioned above.
```
web.ssh_use_salt_thin = true
```

**Comment by juliogonzalez** [comment]
A PR to address this issue (#5139) was merged, and I just re-released Uyuni 2022.03 with the fix included.

The fix should be present at the mirrors in a couple of hours maximum.

**Comment by jorx-dev** [comment]
Hello,

Just updated to the re-released 2022.03 and can confirm Uyuni Server can bootstrap clients again.

**Comment by juliogonzalez** [comment]
Closing.

Everyone having this issue, please check: https://lists.opensuse.org/archives/list/announce@lists.uyuni-project.org/message/ETRNTZRQLDQTAEDP2RKYI2YUMIACP7PF/
